12/hw14phonebook.py

```diff
@@ -0,0 +1,65 @@
+import json
+import re
+phonebook = {"lili":"5555137"}
+data = json.dumps(phonebook)
+with open("phonebook.json", "w") as book:
+    book.write(data)
+
+commands = ["stats", "add", "delete", "list", "show", "exit"]
+while True:
+    com = (input("Please choose your command: stats, add, delete <name>, list, show <name>, exit ")).split()
+    with open("phonebook.json", "r") as book:
+        data = book.read()
+    phonebook = json.loads(data)
+
+    if com[0] not in commands:
+        print("wrong command")
+    else:
+
+        match com[0]:
+            case "stats":
+                print(f"There are {len(phonebook)} numbers in your phone book ")
+
+            case "add":
+                name = input("Please enter new name")
+                if name not in phonebook:
+                    number = input("Please enter phone number")
+                    match = re.search('(\+380|380|0)(\d{9})', number)
```

Comment:
nice, but it will search incorrect phones consisting of more than 9 numbers, so you can add `(?=[, ]|$)` by checking the ending of the phone